### ISSUE TITLE
Compiler crashes with error -2146232797 when attempting to call non-static method as static method on ref struct.

### ISSUE THREAD

**Issue by Mr-Byte** [dotnet/roslyn#23990]
**Version Used**: C# 7.2 (Roslyn 2.4.0)

**Steps to Reproduce**:

The following code causes the compiler to crash when compiling:
```
using System;

namespace Explode
{
    ref struct Test
    {
        public Test Create() => default;
    }

    class Program
    {
        static void Main(string[] args)
        {
            var x = Test.Create();
        }
    }
}
```

**Expected Behavior**:
Should produce compiler error `Program.cs(14,21): error CS0120: An object reference is required for the non-static field, method, or property 'Test.Create()'`

**Actual Behavior**:
Compiler crashes with error `C:\Program Files\dotnet\sdk\2.1.2\Roslyn\Microsoft.CSharp.Core.targets(84,5): error MSB6006: "csc.dll" exited with code -2146232797.`

**Comment by jaredpar** [comment]
Verified this is fixed now.

https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDZICYgNQA+AAgAwAERAjANwCwAUAwHYRhxQAOEAxnGQKIAPDhgD22OAwDeDMnLII4AMzKwEAV24wyAFXYxZ8mfXmmKAZl36yAYUUR4ACgCUZALwA+MhKUR1GGDoTeQBfBkM5IgAmCkoAdgiyYzN5KgA2CgAWWIAORyoSAG0AXTIIBABzKGdE02SUswA3crJBdytYADo7OAc4FyCGsjDguRGQoA
